### DISCUSSION TITLE
RLC-810A Stutter/Lag?

### DISCUSSION THREAD

**Discussion by LostPoE** [thirtythreeforty/neolink#266] [Q&A]
I notice the video feed looks to slow down then speed up to normal speed. Not sure if this is called stutter or lag. Are there any fixes out there for this issue?

The "delay frames" goes from 1 to 6 constantly and you can see the lag in the camera's timestamp.

![image](https://user-images.githubusercontent.com/3904973/162210423-253d6e2e-1ff4-4baf-ac05-4455f6ab5616.png)

Thanks

**Comment by LostPoE** [comment]
I'm thinking this is an RLC-810a issue when using both streams at the same time. Probably hardware.

**Reply by QuantumEntangledAndy** [reply]
Might I ask if you are connecting via `uid` or address in your `neolink.toml`?

If uid then we have a know issue on this caused by udp missing packets that need to be rerequested and then reorganised in a buffer.

**Reply by LostPoE** [reply]
Connecting with IP address.

**Reply by QuantumEntangledAndy** [reply]
I see, not that then.

I should also mention that the time stamps are a bit arbitrary, so that might be effecting your analysis. The actual timestamps are included in the meta data of the custom protocol the camera uses. At one point we had a branch that tried to use these time stamps but it caused all sorts of issues related to simultaneous play on two clients, so it was shelved for later study. Instead we just use the wall clock as the packets are sent out of neolink. So you will be seeing any delays from camera->neolink causing stuttering

**Reply by LostPoE** [reply]
Ah. I could probably live with a little delay. But what I'm focusing on is the stutter. It's like a car going down the road at 30mph but then watching it through Blue Iris or VLC it will slow down to 5mph then speed up to 45mph then back to 30mph then start it all over. Probably my OCD kicking in hah.

**Reply by QuantumEntangledAndy** [reply]
Incorrect time stamps will stutter like that. If a surge of frames comes into the neolink it sends them all off with nearly identical timestamps so it will seem to be moving extra fast on a client. Conversely if frames come in slowly they will get slow timestamps. You need a good solid signal between neolink and the camera to mitigate this. Hopefully I will find time one day to tackle the timestamps properly

**Comment by LostPoE** [comment]
For history. I ended up buying a Reolink NVR today to manage my Reolink RLC-810As. These cameras are not 3rd party NVR friendly (Blue Iris). 

My lessons learned:
1. Don't fully trust "The Hook Up" YouTube Channel on security cam recommendations.
2. Never buy multiple cameras without knowing 1 works.

@QuantumEntangledAndy Thanks for your time.

**Reply by spacebares** [reply]
I never had trouble with 810a. Its been the most solid cam from reolink.
Every other cam however... huge problems left and right. Got turret & dome 8MP they have creeping delay that i fixed by using Neolink. Try feeding blueIris the stream generated from neolink. That is what i do with frigate and it has solved so many problems for me.

Looking back on it if i had known how bad reolink is for 3rd party stuff i would have spent alittle more and got hikvision or amcrest. I too jumped on the reolink bandwagon cause of the hook up.
